docs/configuration.md

```diff
@@ -0,0 +1,140 @@
+# Configuring Gasket
+
+All configuration of the Gasket plugins is centralized in a `gasket.js`
+file in the root of your application.
+
+## Presets and Plugins
+
+To configure plugins in an app, update the `plugins` array in
+`gasket.js` with imported gasket plugin packages.
+
+```js
+import pluginName from '@gasket/plugin-name';
+
+export default makeGasket({
+  plugins: [
+    pluginName
+    /* Plugins to add */ 
+  ],
+  // ...
+});
+```
+
+## Environments
+
+The configuration system supports the concept of
+[environments](https://en.wikipedia.org/wiki/Deployment_environment). An
+application will have certain configuration values that are not
+environment-specific and some, like service URLs, which are. Additionally, an
+environment may be subdivided further in other ways, for example distributed in
+multiple data centers, needing to use specific URLs for DC-local deployments of
+dependencies.
+
+To augment the base `gasket.js` with environment-specific config, add an
+"environments" property to the config object, set to an object. Each key in this
+object should be an environment identifier, and each value is a configuration
+structure to deep merge into the base config. Use `.` in these keys to specify
+sub-environments. For example, here's a theoretical `gasket.js`:
+
+```js
+export default makeGasket({
+  someService: {
+    enableSomeFeature: false
+  },
+  environments: {
+    dev: {
+      someService: {
+        url: 'https://your-dev-service-endpoint.com'
+      }
+    },
+    test: {
+      someService: {
+        url: 'https://your-test-service-endpoint.com'
+      }
+    },
+    prod: {
+      someService: {
+        enableSomeFeature: true
+      }
+    },
+    'prod.datacenter1': {
+      someService: {
+        url: 'https://your-datacenter1-prod-service-endpoint.com'
+      }
+    },
+    'prod.datacenter2': {
+      someService: {
+        url: 'https://your-datacenter2-prod-service-endpoint.com'
+      }
+    }
+  }
+});
+```
+
+When accessing configuration in the `prod.datacenter1` environment, the base
+configuration is augmented with the `prod` environment settings, then the
+`prod.datacenter1` settings, resulting in:
+
+```json
+{
+  "someService": {
+    "enableSomeFeature": true,
+    "url": "https://your-datacenter1-prod-service-endpoint.com"
+  }
+}
+```
+
+The last step is making sure that Gasket is told which environment it is running
+in when starting up. There are three ways of doing this. The first would be to
+set the `GASKET_ENV` environment variable. The second would be to pass an `--env`
+command line argument when running any `gasket` command. The third is to
+programmatically set an `env` property at the top level of your configuration,
+which would enable you to use any arbitrary environment derivation logic.
```

Comment:
The `--env` is no longer an option. We don't support programmatically setting `env`, though perhaps we could?

Looking at [this line](https://github.com/godaddy/gasket/blob/v7/packages/gasket-core/lib/index.js#L54), perhaps we could do something like:
```
const env = gasketConfig.env ?? getEnvironment();
```

Reply:
Should this be in another ticket or is it ok to add to this PR?